src/Security/Authorization/Voter/PublishedVoter.php

```diff
@@ -36,41 +39,27 @@ public function __construct(PublishWorkflowChecker $publishWorkflowChecker)
     /**
      * {@inheritdoc}
      */
-    public function supportsAttribute($attribute)
+    public function supportsAttribute($attribute): bool
     {
         return PublishWorkflowChecker::VIEW_ATTRIBUTE === $attribute
             || PublishWorkflowChecker::VIEW_ANONYMOUS_ATTRIBUTE === $attribute
         ;
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function supportsClass($class)
+    public function supportsType(string $subjectType): bool
     {
-        return $this->publishWorkflowChecker->supportsClass($class);
+        return is_subclass_of($subjectType, PublishableReadInterface::class)
+            || is_subclass_of($subjectType, PublishTimePeriodReadInterface::class);
     }
 
-    /**
-     * {@inheritdoc}
-     *
-     * @param object $subject
-     */
-    public function vote(TokenInterface $token, $subject, array $attributes)
+    protected function supports($attribute, $subject)
     {
-        if (!\is_object($subject) || !$this->supportsClass(\get_class($subject))) {
-            return self::ACCESS_ABSTAIN;
-        }
-        foreach ($attributes as $attribute) {
-            if (!$this->supportsAttribute($attribute)) {
-                return self::ACCESS_ABSTAIN;
-            }
-        }
-
-        if ($this->publishWorkflowChecker->isGranted($attributes, $subject)) {
-            return self::ACCESS_GRANTED;
-        }
+        return $this->supportsType(\get_class($subject))
+            && $this->supportsAttribute($attribute);
+    }
 
-        return self::ACCESS_DENIED;
+    protected function voteOnAttribute($attribute, $subject, TokenInterface $token)
```

Comment:
RE the test failure: not sure if it is a tests fault that it calls this voter with bad data, or if the system would still call us on a $subject that is not supported.

Reply:
yeah, we cant assume its an object